app/controllers/spree/admin/users_controller.rb

```diff
@@ -100,6 +95,16 @@ def collection
 
       private
 
+      def update_message
+        if show_api_key_view_toggled?
```

Comment:
Rails has you covered already:
```suggestion
        if @user.show_api_key_view_previously_changed?
```

It also has the advantage of checking if the value changed after casting to its type which is more accurate than comparing with the raw value from `params`.

Reply:
Okay :) I separated all the api_key messages and used these method for generating the right flash message.